lib/connection.go

```diff
@@ -103,6 +107,11 @@ func (c *Conn) SetHosts(newhosts []string) {
 	c.initializeHostPool()
 }
 
+// Sets the timeout for each individual request.
+func (c *Conn) SetTimeout(timeout time.Duration) {
```

Comment:
this is strange, in that is is a method on the `Conn` struct, but it mutates a shared package variable.